core/store/src/main/java/mil/nga/giat/geowave/core/store/metadata/AbstractGeoWavePersistence.java

```diff
@@ -87,24 +84,30 @@ abstract protected ByteArrayId getPrimaryId(
 			final T persistedObject );
 
 	public void removeAll() {
-		cache.clear();
+		cache.invalidateAll();
 	}
 
 	protected ByteArrayId getCombinedId(
 			final ByteArrayId primaryId,
 			final ByteArrayId secondaryId ) {
 		// the secondaryId is optional so check for null
+		StringBuilder sb = new StringBuilder(
```

Comment:
Earlier in the development of this patch I had made a change that appended all the authorizations to the combined key, making stringbuilder more efficient.  But that was later scrapped and I didn’t fully revert the change.

Reply:
It may be ok, it just requires the IDs to be byte arrays to use a valid character set so they can go back and forth from string encoding without being lossy